scripts/zones/Silver_Sea_route_to_Nashmau/Zone.lua

```diff
@@ -7,26 +7,41 @@ local ID = require("scripts/zones/Silver_Sea_route_to_Nashmau/IDs")
 -----------------------------------
 
 function onInitialize(zone)
-end;
+end
 
 function onZoneIn(player,prevZone)
-    local cs = -1;
-
-    return cs;
-end;
+    local cs = -1
+    -- Gets current vanadiel time and compares it with arrival time to start a timer to show a message before it docks.
+    local vHour = VanadielHour()
+    local vMin  = VanadielMinute()
+    local dockHour
+    local dockMin = 40
+    if vHour < 5 then
+        dockHour = 4
+    elseif vHour < 13 then
+        dockHour = 12
+    elseif vHour < 21 then
+        dockHour = 20
+    end
+    local earthSecsAtZoneIn = math.floor(2.4 * ((vHour * 60) + vMin))
+    local earthSecsAtDocking = math.floor(2.4 * ((dockHour * 60) + dockMin))
+    local earthSecsUntilDock = math.floor(earthSecsAtDocking - earthSecsAtZoneIn)
+    player:timer(math.floor(1000 * earthSecsUntilDock), function(player) player:messageSpecial(ID.DOCKING_IN_NASHMAU) end)
```

Comment:
they do not get the message, i guess timers get reset after zoning.
I tried with !zone gm command but now that you mention it i am going to try using warp spell or ring.

Nice catch =)

Reply:
One thing you might look into is potentially setting a timer on a nearby NPC (invisible or otherwise). It's not like _they're_ going anywhere~

Reply:
@ibm2431  i tried doing:
```lua
    GetNPCByID(17014822):timer(1000 * earthSecsUntilDock, function(player) player:messageSpecial(ID.DOCKING_IN_NASHMAU) end)
```
and it crashes the server. That ID is for the ship captain. Am i doing your suggestion wrong?

Also I was able to confirm that the message doesn't show if the player warps out of the zone before the ship docks.

Let me know otherwise if the timer needs to moved or it can stay on the player, do not see an issue with the timer on the player ^^;;

Reply:
well the player would no longer be there for `player:messageSpecial()` calls to keep working, so that is expected - do the whole thing without the player, make it an area message. Pretty sure that's how retail handles the message while on the boat.